lib/jekyll/site.rb

```diff
@@ -158,7 +164,6 @@ def render
         end
       end
 
-      payload = site_payload
```

Comment:
Not sure I understand this. The same line: ```payload = site_payload``` is just 10 lines above. If it does really need to be here, do you have another suggestion on how we can allow the :site, :pre_render hook to modify the payload?